### DISCUSSION TITLE
Kotlin-based Quarkus Extension

### DISCUSSION THREAD

**Discussion by stephan-strate** [quarkusio/quarkus#31304] [Q&A]
Has anyone experience with kotlin-based extensions? Any pitfalls you are aware of?

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
I am personally not aware of any such extensions. Perhaps you can ask in our [Development mailing list](https://groups.google.com/g/quarkus-dev).

**Comment by RubinCahuegB** [comment]
Kotlin is a modern programming language that has gained popularity in recent years, especially in the Android development community. Kotlin is interoperable with Java, which makes it easy to integrate Kotlin code with Java-based applications. It is also possible to use Kotlin to develop extensions for other programming languages such as C++.

Here are some potential pitfalls to consider when developing Kotlin-based extensions:

Learning Curve: If you are new to Kotlin, there may be a learning curve to overcome. Kotlin has a syntax that is similar to Java, but there are some differences, and it may take some time to become familiar with the language.

Compatibility: Kotlin is a relatively new language, and some libraries and frameworks may not be compatible with it yet. This could limit your options when developing extensions.

Performance: Kotlin is a high-level language, and it may not be as performant as lower-level languages like C++ or Rust. This could be a concern if your extension requires high performance.

Community Support: Kotlin has a growing community, but it may not be as large as other languages like Java or Python. This could limit the availability of resources and support when developing extensions.

Integration: Depending on the programming language or platform you are integrating with, there may be additional challenges when integrating Kotlin code. For example, if you are developing a C++ extension, you may need to use JNI (Java Native Interface) to communicate with the Kotlin code.

Overall, Kotlin is a powerful language that can be used to develop extensions for a variety of programming languages and platforms. However, there may be some challenges and limitations to consider, depending on your specific use case.

**Comment by BurlRooksmhJ** [comment]
well, look who's getting fancy with their Kotlin-based extensions! Sounds like someone's ready to take their programming game to the next level. But be warned, dear friend, with great power comes great responsibility...and sometimes, a few unexpected pitfalls.

But don't worry, you're not the first to dabble in the world of Kotlin-based extensions, so there's plenty of wisdom to be gleaned from those who've come before you. Some common pitfalls to watch out for include issues with type inference, naming conflicts, and even the occasional existential crisis (just kidding...maybe).

But in all seriousness, Kotlin-based extensions can be a great way to enhance your code and make it more readable and maintainable. Just be sure to do your due diligence and test your extensions thoroughly to avoid any unexpected surprises.

So go forth, brave programmer, and explore the world of Kotlin-based extensions! Just remember, with great extensions comes great responsibility...and the occasional bug.

**Comment by DanaEstradeXW9** [comment]
I've actually played around with Kotlin-based extensions before, and for the most part, they're pretty straightforward. That being said, one potential pitfall to watch out for is making sure that you're using the right version of Kotlin for your project, as sometimes there can be compatibility issues with different versions. Other than that, I haven't run into any major issues

**Reply by stephan-strate** [reply]
Did you just add the kotlin-maven-plugin? I feel like, I am missing something, because it does not work for me. Can you maybe provide a minimal working example?

**Reply by b0ro** [reply]
Hello could you elaborate on that? Did you add dependency to kotlin-extension (is it even possible)?
